src/wp-includes/class-wp-term.php

```diff
@@ -182,8 +184,9 @@ public static function get_instance( $term_id, $taxonomy = null ) {
 		}
 
 		$term_obj = new WP_Term( $_term );
-		$term_obj->filter( $term_obj->filter );
-
+		if ( $filter ) {
+			$term_obj->filter( $filter );
+		}
```

Comment:
The previous argument passed to `::filter()` was `$term_obj->filter`, and the new one is `$filter`.

Given a `WP_Term` object with its `$filter` property set to `'rss'`:
```php
$my_term = new WP_Term(
    array(
        'filter' => 'rss',
        // More data.
    )
); // Assume term_id = 1
```
And this call to `get_term()`:

```php
$term = get_term( 1 );
```
Previously, this would have been filtered for `'rss'` in `WP_Term::get_instance()`, then filtered for `'raw'` in `get_term()` before being returned. Now, `WP_Term::get_instance()` won't filter it for `'rss'` before returning to `get_term()`. This could lead to a different return value.

I think we'll need to make sure there's PHPUnit tests covering this scenario and similar ones to make sure we're not implementing a BC break here.

Can pass`'filter' => 'rss'` when creating the term, then hook `term_{$field}_rss` to modify the value, and finally test that calling `get_term()` returns the same value for the given field before/after the patch.

Similarly, using the same term, `get_term()` should be called using `'db'` for the `$filter` parameter, so that neither `get_term()` nor `WP_Term::get_instance()` are using their defaults, followed by testing the field's value before/after the patch.

Reply:
@costdev Could you provide a code example of the issue you documented? I am finding it hard to replicate the above.

So called `get_term` as a parameter for filter, that would be respected. `WP_Term::get_instance` stores term objects in object cache in the raw filter. 

The only issue related here would doing this. 

```
$my_term = new WP_Term(
    array(
        'filter' => 'rss',
    )
);
$my_term::get_instance(123);
```

This would no longer apply the rss filter. The above feels a little edge case, but I may look into it.

Reply:
Sorry for the delayed response! `WP_Term::$filter` not being respected was the issue I was mentioning.

This can be demonstrated by running this test with/without the patch:
```php
public function test_get_term_should_respect_term_filter_property() {
	add_filter(
		'term_name_rss',
		static function() {
			return 'RSS Name';
		}
	);

	$term = new WP_Term(
		(object) array(
			'term_id' => 999,
			'name'    => 'My Test Term Name',
			'filter'  => 'rss',
		)
	);

	// Save to cache for later retrieval.
	wp_cache_set( $term->term_id, $term, 'terms' );

	// Test the name before running `get_term()`.
	$this->assertSame( 'My Test Term Name', $term->name, 'The original term name was not set.' );

	// Retrieve later.
	$actual = get_term( $term->term_id );

	// Test that the filter property has been respected.
	$this->assertSame( 'RSS Name', $actual->name, 'The "rss" filter was not applied despite being the term filter.' );
}
```
The above test is one way to reproduce the issue. There may be other ways in which a term's `$filter` property is set before the term is filtered.

Some notes that might be pertinent when investigating this issue:
- The `$filter` property has `public` visibility.
- `sanitize_term()` will change the `$filter` property to match `$context` value.
- `'rss'` is used just as an example of a non-`'raw'` filter value. The other possible values are:
  - `'edit'`
  - `'db'`
  - `'display'`
  - `'attribute'`
  - `'js'`

Reply:
@costdev That test is invalid. 

Look at the following. 

https://github.com/WordPress/wordpress-develop/blob/da28362ba1a99a86211dfc75c376b16f8218b967/src/wp-includes/class-wp-term.php#L176-L181

As you can see the term object stored as filter = raw. There is never a case where it would be stored filtered for rss.